Sources/NIO/Codec.swift

```diff
@@ -222,13 +222,13 @@ extension ByteToMessageDecoder {
     public func shouldReclaimBytes(buffer: ByteBuffer) -> Bool {
         // We want to reclaim in the following cases:
         //
-        // 1. If there is more than 2kB of memory to reclaim
+        // 1. If there is at least 2kB of memory to reclaim
         // 2. If the buffer is more than 50% reclaimable memory and is at least
         //    1kB in size.
-        if buffer.readerIndex > 2048 {
+        if buffer.readerIndex >= 2048 {
```

Comment:
what's the motivation for this change? Seems unrelated to the bugfix and obfuscates the important change?

Reply:
In `CodecTests` we currently explicitly test this behavior. The test succeeded up until now, since it used the buggy code path:
```swift
    func testMemoryIsReclaimedIfLotsIsAvailable() throws {
        let channel = EmbeddedChannel()
        defer {
            XCTAssertNoThrow(try channel.finish())
        }

        let decoder = ByteToMessageHandler(OnceDecoder()) // <--- the once decoder consumes exactly 2048 bytes.
        _ = try channel.pipeline.addHandler(decoder).wait()

        // We're going to send in 5119 bytes. This will be held.
        var buffer = channel.allocator.buffer(capacity: 5119)
        buffer.writeBytes(Array(repeating: 0x04, count: 5119))
        XCTAssertTrue(try channel.writeInbound(buffer).isEmpty)

        XCTAssertEqual(decoder.cumulationBuffer!.readableBytes, 5119)
        XCTAssertEqual(decoder.cumulationBuffer!.readerIndex, 0)

        // Now we're going to send in one more byte. This will cause a chunk to be passed on,
        // shrinking the held memory to 3072 bytes. However, memory will be reclaimed.
        XCTAssertTrue(try channel.writeInbound(buffer.getSlice(at: 0, length: 1)).isFull)
        XCTAssertEqual(decoder.cumulationBuffer!.readableBytes, 3072)
        XCTAssertEqual(decoder.cumulationBuffer!.readerIndex, 0)
    }
```

Reply:
Thank you! That makes sense.

Reply:
offline discussion with Fabian:

- the change in this logic is kinda arbitrary but it also won't hurt
- the documentation is just a code comment and _not_ in the public API docs so we never guaranteed what we'd do anyway so shifting the arbitrary limit by 1 byte shouldn't matter